test/integration/master/kube_apiserver_test.go

```diff
@@ -85,6 +85,83 @@ func TestRun(t *testing.T) {
 	}
 }
 
+type delayedCheck struct {
+	startPassingAfter time.Time
+}
+
+func (h delayedCheck) Name() string {
+	return "fail-for-ten-seconds"
+}
+
+func (h delayedCheck) Check(req *http.Request) error {
+	if time.Now().After(h.startPassingAfter) {
+		return nil
+	}
+	return fmt.Errorf("failing until after %v", h.startPassingAfter.Unix())
+}
+
+func TestDelayedHealthzAndReadyz(t *testing.T) {
+	hc := delayedCheck{time.Now().Add(10 * time.Second)}
```

Comment:
why not make that synchronously controlled? I.e. put a protected failing bool inside and then set that during startup and later after startup.

Reply:
Also a ten second runtime of a test is not good. This should be as fast as possible. Use `wait.ForeverTestTimeout` as upper limit everywhere, but otherwise it should run without fixed duration values.

Reply:
sure